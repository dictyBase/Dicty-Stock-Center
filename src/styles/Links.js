import styled from "styled-components"

export const LinkedList = styled.div`
  ul,
  li {
    margin: 0;
    padding: 0;
    list-style: none;
  }
  a {
    font-size: 18px;
  }
`

export const InlineLink = styled.a`
  cursor: pointer;
`

export const SopLink = styled.a`
  color: #961818;
`
