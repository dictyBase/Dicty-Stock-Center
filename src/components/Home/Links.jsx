import React from 'react'
import LinkList from '../LinkList'
import { PanelBlue } from 'styles'

const content = [
  { name: 'Contact the DSC', to: '/contact', routerAware: true },
  { name: 'DSC FAQ', to: '/information/faq', routerAware: true },
  {
    name: 'Nomenclature Guide',
    to: '/information/nomenclature',
    routerAware: false
  },
  {
    name: 'Other Stock Centers',
    to: '/information/other-stock-centers',
    routerAware: true
  }
]

const Links = () => {
  Links.displayName = 'front page links'
  return (
    <PanelBlue>
      <LinkList list={content} />
    </PanelBlue>
  )
}

export default Links
