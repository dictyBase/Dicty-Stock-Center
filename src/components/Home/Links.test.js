import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Links from './Links'

test('matching a snapshot of Links', () => {
    const component = renderer.create(
    <BrowserRouter>
        <Links />
    </BrowserRouter>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
