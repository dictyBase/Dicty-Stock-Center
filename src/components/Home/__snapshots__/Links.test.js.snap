// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Links 1`] = `
<div
  className="sc-fzsDOv bjlbND"
>
  <div
    className="sc-gPWkxV dfohwQ"
  >
    <ul>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact the DSC
        </a>
      </li>
      <li>
        <a
          href="/information/faq"
          onClick={[Function]}
        >
          DSC FAQ
        </a>
      </li>
      <li>
        <a
          href="/information/nomenclature"
        >
          Nomenclature Guide
        </a>
      </li>
      <li>
        <a
          href="/information/other-stock-centers"
          onClick={[Function]}
        >
          Other Stock Centers
        </a>
      </li>
    </ul>
  </div>
</div>
`;
